Userland/Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -207,10 +206,12 @@ void HTMLMediaElement::set_duration(double duration)
     // is not fired when the duration is reset as part of loading a new media resource.) If the duration is changed such that the current playback position
     // ends up being greater than the time of the end of the media resource, then the user agent must also seek to the time of the end of the media resource.
     if (!isnan(duration)) {
-        // FIXME: Handle seeking to the end of the media resource when needed.
         queue_a_media_element_task([this] {
             dispatch_event(DOM::Event::create(realm(), HTML::EventNames::durationchange).release_value_but_fixme_should_propagate_errors());
         });
+
+        if (m_current_playback_position > duration)
+            seek_element(duration);
```

Comment:
As of right now this is actually unreachable anyways. We only enter `set_duration` with a valid duration just after creating a `PlaybackManager`, and we set `m_current_playback_position` to 0 just before.

There's probably other spec steps we haven't implemented yet that would cause this to be reached, so when we get there, we'll see how it behaves I guess

Reply:
I figured that was probably the case, no big deal for now. :^)